install.sh

```diff
@@ -1344,43 +1395,127 @@ update_release_clone() {
         GITHOOKS_CLONE_BRANCH=$(git config --global githooks.autoupdate.updateCloneBranch)
     fi
 
-    if is_git_repo "$CLONE_DIR"; then
-        URL=$(git -C "$CLONE_DIR" config remote.origin.url)
-        BRANCH=$(git -C "$CLONE_DIR" symbolic-ref -q --short HEAD)
+    if is_git_repo "$GITHOOKS_CLONE_DIR"; then
+
+        URL=$(execute_git "$GITHOOKS_CLONE_DIR" config remote.origin.url 2>/dev/null)
+        BRANCH=$(execute_git "$GITHOOKS_CLONE_DIR" symbolic-ref -q --short HEAD 2>/dev/null)
 
         if [ "$URL" != "$GITHOOKS_CLONE_URL" ] ||
-            [ "$BRANCH" != "$GITHOOKS_CLONE_BRANCH" ] ||
-            ! git -C "$CLONE_DIR" diff-index --quiet HEAD; then
-            PULL_ONLY="false"
+            [ "$BRANCH" != "$GITHOOKS_CLONE_BRANCH" ]; then
+
+            CREATE_NEW_CLONE="true"
+
+            # During an autoupdate we would silently erase/reclone
+            # which is not so good, therefore we abort here.
+            if is_autoupdate; then
```

Comment:
Will this work for people updating to this version?
Do we already have those variables set to the right branch and URL?

Reply:
From really really old scripts we probably not pass the test `if is_autoupdate` because the update doesnt pass the flag to install.sh. But should not matter we then just clone from **this** (default url) repo as a default. For other version already passing the flag to `install.sh` it should work.
(As a sidenote we should maybe think about deprecating update from too old version, to get our head clean of crazy thoughts about what all can happen, I dont know yet a strategy to detect from where we are updating... to deny the update if we are too old and then asking the user to uninstall and reinstall a latest version... etc)